### PR TITLE
Fix flaky completions for `zsh`

### PR DESCRIPTION
**Pull request by Jasper-M**
Quotes break zsh completions.

Or more specifically, escaped quotes `\'` seem to break completions on zsh.

### CODE REVIEW

**Comment by Gedochao** [comment]
Hey, thanks for the contribution!
I tried to replicate the problem with completions  and `zsh` and ran into something like this:
```
▶ scala-cli 
(eval):37: unmatched '
(eval):37: unmatched '
(eval):37: unmatched '
```
(`scala-cli ` with a space and then `TAB`, breaks)
```
▶ scala-cli c
clean    -- Clean the workspace.
compile  -- Compile Scala code.
config   -- Configure global settings for Scala CLI.
console  -- Fire-up a Scala REPL.
```
(adding a `c` brings the completions back)

Is this the behaviour you're trying to fix?

**Comment by Gedochao** [comment]
We might need to add a relevant anti-regression test.
The completions integration tests are here: https://github.com/Gedochao/scala-cli/blob/main/modules/integration/src/test/scala/scala/cli/integration/CompleteTests.scala#L7

Do you have the time to add a test?
If not, I can try to look at this later.

**Comment by Gedochao** [comment]
the `reference-doc` check is failing on the CI, btw.
This is because reference docs haven't been regenerated in sync with your changes.

You can fix it by running
```bash
./mill -i generate-reference-doc.run
```
(as per [CONTRIBUTING.md](https://github.com/VirtusLab/scala-cli/blob/main/CONTRIBUTING.md))

**Comment by Jasper-M** [comment]
> (adding a c brings the completions back)

Indeed, once you manage to limit the output to completions that don't contain quotes in their description it works.

> Is this the behaviour you're trying to fix?

Yes that's exactly it. Turns out a similar change has already been attempted in the past https://github.com/VirtusLab/scala-cli/pull/893

> Do you have the time to add a test?

I will give it a go.

I just noticed that completions are implemented by case-app instead of natively in scala-cli. I should probably open an issue there.

**Comment by Jasper-M** [comment]
> I just noticed that completions are implemented by case-app instead of natively in scala-cli. I should probably open an issue there.

alexarchambault/case-app#475

File: modules/integration/src/test/scala/scala/cli/integration/CompleteTests.scala [hunk 1]
```diff
@@ -14,6 +14,15 @@ class CompleteTests extends ScalaCliSuite {
     }
   }
 
+  test("zsh bug") {
+    // guard against https://github.com/alexarchambault/case-app/issues/475
+    TestInputs.empty.fromRoot { root =>
+      val res = os.proc(TestUtil.cli, "complete", "zsh-v1", "2", "scala-cli").call(cwd = root)
```
Comment (Gedochao) [inline modules/integration/src/test/scala/scala/cli/integration/CompleteTests.scala]:
```suggestion
      val res = os.proc(TestUtil.cli, "complete", shellFormat, "2", "scala-cli").call(cwd = root)
```
won't hurt to enforce this to work the same on `bash`.

**Review by Gedochao — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**